### PR TITLE
Optimize the source code and trying to add some functions to save a file.

### PR DESCRIPTION
**Pull request by awesometic**
Please review them and merge.

I did make classes which seems it's instance always being only one at once as a singleton class.
TripManager, Monitor, Observer, RouteTracker... and maybe more I don't remember.

And I added two functions operating disk I/O jobs between our app and the device. But, I think I can't test more at my side due to the simulator which I'm using it for my testing device.

Thanks.

### CODE REVIEW

File: map/tripManager.swift [hunk 1]
```diff
@@ -8,38 +8,46 @@
...
-        }
-        if let event = event {
-            return event
-        } else {
-            return nil
-        }
+    func createCommit(type: eventType, with member: [Member], media: Media?) -> commit? {
+        let event: commit?
+        
+        event = {
```
Comment (xiane) [inline map/tripManager.swift:30]: what's differ from before?

File: map/ViewController.swift [hunk 2]
```diff
@@ -22,22 +22,53 @@ class ViewController: UIViewController {
...
+            if !upTitle.isEmpty && !downTitle.isEmpty {
+                alert.addAction(UIAlertAction(title: upTitle, style: .default,
+                    handler: { action in
+                        DispatchQueue.main.async {
+                            self.share(url: SaveFileTest.share.fileShareURL)
+                        }
+                }))
+                alert.addAction(UIAlertAction(title: downTitle, style: .default,
+                    handler: { action in
+                        SaveFileTest.share.saveFile()
```
Comment (xiane) [inline map/ViewController.swift:50]: this line is look like a point. other alert method lines are looks just tutorial things.

File: map/tripManager.swift (hunk 1 above)
Comment (joshua-yang) [reply map/tripManager.swift:30]: It's intended to do upcasting from descendants(Start, Route, ...) to its super class Commit.
But not sure really. Might need to be changed.

Comment (joshua-yang) [reply map/tripManager.swift:30]: Oh, and I temporary changed define keyword "var" to "let" and assigned a value looks more clearly.

**Review by joshua-yang — COMMENTED** [review]
// I clicked something by mistake :confused:

File: map/ViewController.swift (hunk 2 above)
Comment (joshua-yang) [reply map/ViewController.swift:50]: Yes it seems but it is written only for "development", not further used you know :)

**Comment by xiane** [comment]
I'm going to close the pr because it is not separated by functionality.
and some point of commit should be changed.

**Pull request closed without merging**